doc/developers/porting.html

```diff
@@ -118,40 +118,101 @@ <h2>Porting to a new platform</h2>
-      href="http://www.cmake.org/Wiki/CMake_Cross_Compiling">CMake
-      wiki</a>.
+      It is possible to use the
+      <a href="https://cmake.org/cmake/help/latest/manual/cmake-toolchains.7.html">CMake toolchain</a>
+      to cross compile deal.II for a platform other than the one on which the
+      compiler is running. The target platform can have a different operating
+      system, different architecture or different set of libraries. Cross
+      compilation is a very useful technique, for instance it can be used to
+      compile deal.II with a compiler that is not available in the target
+      machine. An alternative technique is to use a
+      <a href="https://www.docker.com/">Docker container</a> or a
+      <a href="https://en.wikipedia.org/wiki/Virtualization">virtual machine</a>
+      that mimics the target machine.
     </p>
 
     <p>
-      You have to set up a native deal.II build directory first and run
-      <code>make expand_instantiations_exe</code> in it. The executable is
-      needed for the build system (and obviously the cross compiled version
-      cannot be used). Locate the <code>expand_instantiations</code>
-      executable (it usually resides under
-      <code>${CMAKE_BINARY_DIR}/bin</code> and export its location with the
-      <code>PATH</code> environment variable.
+      You can use any compiler for cross compilation, although we find that
+      <a href="https://clang.llvm.org/">LLVM/clang</a> is the most versatile
+      because it is natively a cross-compiler. Below you can find an example
+      toolchain file for cross compilation with clang (and another example for
+      <a href="Toolchain-x86_64-w64-mingw32.sample">Windows64 using MinGW</a>).
+<pre>
+set(target_root /path/to/sysroot)
+set(dealii_dir ${target_root}/path/to/lib/dealii)
+
+set(CMAKE_SYSTEM_NAME Linux)
+set(CMAKE_SYSTEM_PROCESSOR x86_64)
+
+set(CMAKE_SYSROOT ${target_root})
+
+set(CMAKE_C_COMPILER clang)
+set(CMAKE_CXX_COMPILER clang++)
+
+set(CMAKE_C_COMPILER_EXTERNAL_TOOLCHAIN ${target_root}/path/to/gcc/toolchain)
+set(CMAKE_CXX_COMPILER_EXTERNAL_TOOLCHAIN ${target_root}/path/to/gcc/toolchain)
+
+set(CMAKE_FIND_ROOT_PATH_MODE_PROGRAM NEVER)
+set(CMAKE_FIND_ROOT_PATH_MODE_LIBRARY ONLY)
+set(CMAKE_FIND_ROOT_PATH_MODE_INCLUDE ONLY)
+set(CMAKE_FIND_ROOT_PATH_MODE_PACKAGE ONLY)
+
+# You may have to include some c++ headers of the target
+include_directories(SYSTEM ${target_root}/path/to/c++/include/files)
+</pre>
+      A great advantage of LLVM/clang is that you can use the toolchain of the
+      target with the option <code>CMAKE_CXX_COMPILER_EXTERNAL_TOOLCHAIN</code>;
+      this means that you do not have to install a specific clang compiler for
+      cross compiling. On the other hand if you use gcc and the version of the
+      host glibc is higher than the version of the target glibc you may run into
+      undefined reference errors. You should place all the relevant libraries of
```

Comment:
This statement is not correct: The forward compatibility issue with glibc is something you will face with clang as well - the C library is independent of the compiler. Generally, as long as you are not using forward compatible libraries (such as for example musl as C library, or LLVM's libc++ as C++ standard template library) you have to make sure that the glibc and libstdc++ versions used for cross compilation are the same or older than the target.